src/reanimated2/layoutReanimation/web/componentUtils.ts

```diff
@@ -135,6 +135,13 @@ export function getProcessedConfig(
   };
 }
 
+export function saveSnapshot(element: HTMLElement) {
+  element.setAttribute(
+    'data-reanimated-snapshot',
+    JSON.stringify(element.getBoundingClientRect())
+  );
```

Comment:
We could also store these snapshots in a `WeakMap`, this way we don't need to serialize and parse it.

Reply:
Done in [bedb8f7](https://github.com/software-mansion/react-native-reanimated/pull/5444/commits/bedb8f73bed613dc4842082d807dd388640e2a6a)